### DISCUSSION TITLE
Is it possible to provide a minimal custom HTTPHandler with TypeScript?

### DISCUSSION THREAD

**Discussion by vidjul** [mswjs/msw#1917] [Q&A] [answered]
Hey there! 👋

I'm trying to extending the RequestHandler to add some extra properties on the class but I'm struggling with the typing.

Is it possible to provide me a minimal example of a custom handler like the one https://mswjs.io/docs/api/request-handler#properties with the proper types?

```
class MyHandler extends RequestHandler {
  constructor(method, path, resolver) {
    super({
      info: {
        // Public representation of this request handler.
        // This string will be used when logging the handler
        // using the ".log()" method.
        header: `${method} ${path}`,
        // Additional info properties forwarded from the
        // constructor arguments of this custom handler.
        method,
        path,
      },
      resolver,
    })
  }
}
```

Out of topic but the extra property I want to add is the mock associated to the handler. Is it possible to get it without going through a custom handler perhaps?

**Comment by mattcosta7** [comment]
> class MyHandler extends RequestHandler {

I would look at the typings applied for `HttpHandler` and `GraphQLHandler` internally, and that should help you write a custom one based on the base `RequestHandler`

**Reply by mattcosta7** [reply, marked as answer]
https://github.com/mswjs/msw/blob/c27c32000d64bb285998597deacbfc8349f08a55/src/core/handlers/HttpHandler.ts#L59-L63

**Reply by vidjul** [reply]
Your point made me think that actually, I've just wanted to extend `HttpHandler` and not the abstract class that it extends.
All sorted while directly extending `HttpHandler` to add my property, thanks!
